apple/Sources/Connlib/CallbackHandler.swift

```diff
@@ -23,71 +23,47 @@ public class CallbackHandler {
 
     func onUpdateResources(resourceList: ResourceList) -> Bool {
 
-        // If there's any entity that assigned itself as this callbackHandler's delegate, it will be called everytime this `onUpdateResources` method is, allowing that entity to react to resource updates and do whatever they want.
-
-        delegate?.didUpdateResources(resourceList)
-
-        let addresses4 =
-        self.adapter?.lastNetworkSettings?.ipv4Settings?.addresses ?? ["100.100.111.2"]
-        let addresses6 =
-        self.adapter?.lastNetworkSettings?.ipv6Settings?.addresses ?? [
-            "fd00:0222:2021:1111::2"
-        ]
+        let logger = Logger(subsystem: "dev.firezone.firezone", category: "packet-tunnel")
 
-        // TODO: Use actual passed in resources to achieve split tunnel
-        let ipv4Routes = [NEIPv4Route(destinationAddress: "100.64.0.0", subnetMask: "255.192.0.0")]
-        let ipv6Routes = [
-            NEIPv6Route(destinationAddress: "fd00:0222:2021:1111::0", networkPrefixLength: 64)
-        ]
+        do {
+            let resources = try resourceList.toResources()
+            logger.debug("Resources updated: \(resources)")
+            delegate?.didUpdateResources(resources)
+        } catch {
+            logger.error("Error parsing resource list: \(String(describing: error)) (JSON: \(resourceList.resources.toString()))")
+            return false
+        }
 
-        return setTunnelSettingsKeepingSomeExisting(
-            addresses4: addresses4, addresses6: addresses6, ipv4Routes: ipv4Routes, ipv6Routes: ipv6Routes
-        )
+        return true
     }
 
     func onSetTunnelAddresses(tunnelAddresses: TunnelAddresses) -> Bool {
-        let addresses4 = [tunnelAddresses.address4.toString()]
-        let addresses6 = [tunnelAddresses.address6.toString()]
-        let ipv4Routes =
-        Adapter.currentAdapter?.lastNetworkSettings?.ipv4Settings?.includedRoutes ?? []
-        let ipv6Routes =
-        Adapter.currentAdapter?.lastNetworkSettings?.ipv6Settings?.includedRoutes ?? []
-
-        return setTunnelSettingsKeepingSomeExisting(
-            addresses4: addresses4, addresses6: addresses6, ipv4Routes: ipv4Routes, ipv6Routes: ipv6Routes
-        )
-    }
-
-    private func setTunnelSettingsKeepingSomeExisting(
-        addresses4: [String], addresses6: [String], ipv4Routes: [NEIPv4Route], ipv6Routes: [NEIPv6Route]
-    ) -> Bool {
         let logger = Logger(subsystem: "dev.firezone.firezone", category: "packet-tunnel")
 
-        if self.adapter != nil {
-            do {
-                /* If the tunnel interface addresses are being updated, it's impossible for the tunnel to
-                 stay up due to the way WireGuard works. Still, we try not to change the tunnel's routes
-                 here Just In Case™.
-                 */
```

Comment:
Maybe not. I think we can update the routes without bringing down the tunnel, but I'm not sully sure because I haven't tried it in practice.

[Documentation](https://developer.apple.com/documentation/networkextension/netunnelprovider/1406539-settunnelnetworksettings#parameters) and [forums](https://developer.apple.com/forums/thread/659298) suggest we are allowed to call setTunnelNetworkSettings() while the tunnel is up. Have not checked it in practice.